build-system/tasks/dist.js

```diff
@@ -0,0 +1,375 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const $$ = require('gulp-load-plugins')();
+const colors = require('ansi-colors');
+const fs = require('fs-extra');
+const gulp = $$.help(require('gulp'));
+const log = require('fancy-log');
+const {
+  buildAlp,
+  buildExaminer,
+  buildExperiments,
+  buildWebWorker,
+  compile,
+  compileJs,
+  enableLocalTesting,
+  endBuildStep,
+  hostname,
+  mkdirSync,
+  printConfigHelp,
+  toPromise,
+} = require('./helpers');
+const {
+  buildExtensions,
+  extensionAliasFilePath,
+  getExtensionsToBuild,
+  parseExtensionFlags,
+} = require('./extension-helpers');
+const {cleanupBuildDir} = require('../compile/compile');
+const {closureNailgunPort, startNailgunServer, stopNailgunServer} = require('./nailgun');
+const {compileCss, cssEntryPoints} = require('./css');
+const {createCtrlcHandler, exitCtrlcHandler} = require('../ctrlcHandler');
+const {createModuleCompatibleES5Bundle} = require('./create-module-compatible-es5-bundle');
+const {isTravisBuild} = require('../travis');
+
+const {green, cyan} = colors;
+const argv = require('minimist')(process.argv.slice(2));
+const maybeUpdatePackages = isTravisBuild() ? [] : ['update-packages'];
+
+// Minified targets to which AMP_CONFIG must be written.
+const minifiedRuntimeTarget = 'dist/v0.js';
+const minifiedShadowRuntimeTarget = 'dist/shadow-v0.js';
+const minifiedAdsTarget = 'dist/amp4ads-v0.js';
+// TODO(#18934, erwinm): temporary fix.
+//const minifiedRuntimeEsmTarget = 'dist/v0-esm.js';
+const minified3pTarget = 'dist.3p/current-min/f.js';
+
+/**
+ * Dist Build
+ * @return {!Promise}
+ */
+function dist() {
+  const handlerProcess = createCtrlcHandler('dist');
+  process.env.NODE_ENV = 'production';
+  cleanupBuildDir();
+  if (argv.fortesting) {
+    let cmd = 'gulp dist --fortesting';
+    if (argv.single_pass) {
+      cmd = cmd + ' --single_pass';
+    }
+    printConfigHelp(cmd);
+  }
+  if (argv.single_pass) {
+    if (!isTravisBuild()) {
```

Comment:
This nested conditional is the only item inside a conditional, they could be combined.

`if(argv.single_pass && !isTravisBuild())`